core/cap-0066.md

```diff
@@ -355,31 +355,56 @@ should contain no classic key indexes, as classic keys are always considered dis
 Whenever `InvokeHostFunctionOp` is applied, any archived state is
 automatically restored prior to host function invocation. This restored state is then accessible to the host function.
 
-Restored state is still subject to the same minimum rent and write fees that exist currently. All entries being restored
-count towards disk read resources and fees. Even if an entry is archived but not yet evicted such that it technically still
+Restored state is still subject to the same minimum rent and write fees that exist currently based on the final result of the
+invocation. Even if an entry is archived but not yet evicted such that it technically still
 exists in memory, it is still subject to the same limits and fees as disk based entries in order ot provide a simpler unified
 interface for downstream systems.
 
 All archived keys must be declared in the `archivedEntries` vector via `SorobanResourcesExtV0`.
 If no archived entries are present in the footprint, `SorobanResourcesExtV0` may be omitted.
 
+##### Fees
+
+Given that restoration and invocation are happening atomically, rent fees will be charged based on the final
+state following invocation, not the initial restore state. For example, if an entry is autorestored, then deleted
+as part of the invocation, it will only pay disk read fees, not write fees or rent fees.
+
+Suppose we are auto restoring some entry e:
+
+```
+min_ttl = minimum required ttl following restore
+e = initial value of ledger entry that is restored
+e' = value of e following transaction execution
+e'_ttl = ttl value of e following transaction execution
+
+Metering and Fees for InvokeHostFunctionOp with autorestore of e:
+
+diskReadBytes = sizeof(e)
+writeBytes = sizeof(e')
+
+if sizeof(e') == 0:
+     rent_fee = 0
+else:
+     rent_fee = rent_fee_for_size_and_ledgers(/*isPersistent*/ true, sizeof(e'), max(min_ttl, e'_ttl))
+```
+
 ##### Incorrect Entries in `archivedEntries` Vector
 
 If a key is not marked as archived, but the entry is actually archived, then the TX fails. If a Soroban key is marked as archived when
 it is not, the transaction does not fail. The entry is treated as if it is archived from a resource and fee perspective.
 This is appropriate, as on disk reads are more expensive than in-memory. So long as the TX pays the disk based fees,
 there is no issue with actually loading an in-memory entry.
 
-If a classic key is marked as archived, the TX fails, as the footprint is malformed.
+If `archivedSorobanEntries` contains an invalid index (either an out of bounds index or referring to a classic entry)
+the transaction is malformed and will be dropped from the transaction queue. Indices in `archivedSorobanEntries` must be sorted.
```

Comment:
We shouldn't refer to tx queue in the CAPs as it's a non-protocol implementation detail. Just 'is malformed' seems  to be enough.